src/main/java/payapal/Perfection.java

```diff
@@ -0,0 +1,42 @@
+package main.java.payapal;
+
+public class Perfection {
+
+    private static Perfection perf;
+
+    public synchronized static Perfection getPerf() {
+        if (perf == null) {
+            perf = new Perfection();
+        }
+        return perf;
+    }
+
+
+    public static boolean isPerfect(long candidate) {
+        boolean retVal;
```

Comment:
Variable was not initialized here and it will lead to a compile time error. Initialize it with false. 
Also, you could look into good naming conventions for boolean variables. `isPerfect` would be a good fit here.